packages/snip20/src/query.rs

```diff
@@ -0,0 +1,315 @@
+use core::fmt;
+use schemars::JsonSchema;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+
+use cosmwasm_std::{
+    to_binary, Api, Coin, Extern, HumanAddr, Querier, QueryRequest, StdError, StdResult, Storage,
+    Uint128, WasmQuery,
+};
+
+use secret_toolkit_utils::space_pad;
+
+/// TokenInfo response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct TokenInfo {
+    pub name: String,
+    pub symbol: String,
+    pub decimals: u8,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub total_supply: Option<Uint128>,
+}
+/// ExchangeRate response
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct ExchangeRate {
+    pub rate: Uint128,
+    pub denom: String,
```

Comment:
please add an empty line between each two top-level items. (the structs, not their fields)